traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -0,0 +1,155 @@
+	hostName := ""
+	if err == nil {
+		name, ok, err := dbhelpers.GetServerNameFromID(inf.Tx.Tx, id)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting server name from id '"+idOrName+"': "+err.Error()))
+			return
+		} else if !ok {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("server ID '"+idOrName+"' not found"), nil)
+			return
+		}
+		hostName = name
+	} else {
+		hostName = idOrName
+		if _, ok, err := dbhelpers.GetServerIDFromName(hostName, inf.Tx.Tx); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting server id from name '"+idOrName+"': "+err.Error()))
+			return
+		} else if !ok {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("server name '"+idOrName+"' not found"), nil)
+			return
+		}
+	}
+
+	updated, hasUpdated := inf.Params["updated"]
+	revalUpdated, hasRevalUpdated := inf.Params["reval_updated"]
+	if !hasUpdated && !hasRevalUpdated {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("Must pass at least one query paramter of 'updated' or 'reval_updated'"), nil)
+		return
+	}
+	updated = strings.ToLower(updated)
+	revalUpdated = strings.ToLower(revalUpdated)
+
+	if hasUpdated && !strings.HasPrefix(updated, "t") && !strings.HasPrefix(updated, "f") {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'updated' must be 'true' or 'false'"), nil)
+		return
+	}
+	if hasRevalUpdated && !strings.HasPrefix(revalUpdated, "t") && !strings.HasPrefix(revalUpdated, "f") {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'reval_updated' must be 'true' or 'false'"), nil)
+		return
+	}
+
+	strToBool := func(s string) bool {
+		return !strings.HasPrefix(strings.ToLower(s), "f")
+	}
+
+	updatedPtr := (*bool)(nil)
+	if hasUpdated {
+		updatedBool := strToBool(updated)
+		updatedPtr = &updatedBool
+	}
+	revalUpdatedPtr := (*bool)(nil)
+	if hasRevalUpdated {
+		revalUpdatedBool := strToBool(revalUpdated)
+		revalUpdatedPtr = &revalUpdatedBool
+	}
+
+	if err := setUpdateStatuses(inf.Tx.Tx, hostName, updatedPtr, revalUpdatedPtr); err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting updated statuses: "+err.Error()))
+		return
+	}
+
+	err = api.CreateChangeLogBuildMsg(
+		api.ApiChange,
+		api.Updated,
+		inf.User,
+		inf.Tx.Tx,
+		"server-update-status",
+		hostName,
+		map[string]interface{}{"host_name": hostName},
+	)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("writing changelog: "+err.Error()))
+		return
+	}
+
+	respMsg := "successfully set server '" + hostName + "'"
+	if hasUpdated {
+		respMsg += " updated=" + strconv.FormatBool(strToBool(updated))
+	}
+	if hasRevalUpdated {
+		respMsg += " reval_updated=" + strconv.FormatBool(strToBool(revalUpdated))
+	}
+	api.WriteResp(w, r, respMsg)
```

Comment:
Success messages should go in `success`-level alerts, not response objects.

Reply:
Done